### ISSUE TITLE
Add aop_add_afterreturning

### ISSUE THREAD

**Issue by kdambekalns** [AOP-PHP/AOP#17]
It would be nice to be able to declare advice only after a method returns normally, using `aop_add_afterreturning`. The existing `aop_add_after()` would be a "superset" of this, so to speak… See http://flow3.typo3.org/documentation/guide/partiii/aspectorientedprogramming.html#after-returning-advice for the FLOW3 documentation on that feature.

**Comment by geraldcroes** [comment]
First, thx for all your requests, we really appreciate feedback.

Currently aop_add_after does not trigger while an exception is launched, so it basically act as aop_add_afterreturning in Flow3.

We had in mind the "aop_add_finally" that would deal with your request, but wanted to release the beta 1 sooner, so we postponed this feature.

Julien & I discussed your request and it should be planned for beta 2.

Does it seems to adress your need ?

**Comment by kdambekalns** [comment]
That would be fine. We are not in a hurry, I was just looking at the extension today to see what it lacks compared to FLOW3 - because if we can integrate it and see a speedup that would be very welcome, of course.

Glad to see you are open to suggestions - focus on your beta 1 and maybe we can play around with what features you have right now in the meantime.

**Comment by geraldcroes** [comment]
See #18

**Comment by geraldcroes** [comment]
Done & tested

e67777e5ed844961a131cd1e06a28f37943e12e6
